src/main/java/org/candlepin/controller/ContentManager.java

```diff
@@ -156,20 +156,36 @@ public Content createContent(Owner owner, ContentInfo contentData) {
             .get(entity.getId());
 
         if (alternateVersions != null) {
-            log.debug("Checking {} alternate content versions", alternateVersions.size());
+            // Impl note: this should only ever have 1 entry in it due to our implicit limitation
+            // on ID and entity version
+            if (alternateVersions.size() != 1) {
+                log.warn("Unexpected number of alternate versions received for content ID {}: {}",
+                    entity.getId(), alternateVersions.size());
+            }
+            else {
+                log.debug("Checking {} alternate content versions", alternateVersions.size());
+            }
 
-            for (Content alt : alternateVersions) {
-                if (!alt.equals(entity)) {
-                    String errmsg = String.format("Entity version collision detected: %s != %s", alt, entity);
-                    throw new IllegalStateException(errmsg);
+            for (Content candidate : alternateVersions) {
+                if (entity.equals(candidate)) {
+                    // We found a match! Map to the candidate entity
+                    log.debug("Converging with existing content version: {} => {}", entity, candidate);
+
+                    // If we're "creating" a content, we shouldn't have any other object references to
+                    // update for this content. Instead, we'll just add the new owner to the content.
+                    this.ownerContentCurator.mapContentToOwner(candidate, owner);
+                    return candidate;
                 }
 
-                log.debug("Converging content with existing version: {} => {}", entity, alt);
+                // If we have a version collision, and the entity IDs are the same, there's likely
+                // some shenanigans going on. Rather than halting all behavior, let's just clear
+                // the old entity's version so we start mapping to the new one.
+                // If we have a collision where the contents are actually different, we won't
+                // fail at this point (but we *will* fail), and we won't be able to detect it.
```

Comment:
Where (and in what way) would we fail if this happened?

Reply:
When a flush is performed at some later point, Hibernate will throw an exception. I've been wondering if the restriction should fall back to unique on ID + entity version to avoid this type of thing all together.